aptos-move/framework/aptos-framework/sources/object.move

```diff
@@ -0,0 +1,338 @@
+        guid_creation_num: u64,
+    }
+
+    #[resource_group_container]
+    struct ObjectGroup { }
+
+    /// This is a one time ability given to the creator to configure the object as necessary
+    struct CreatorAbility has drop {
+        self: address,
+    }
+
+    /// The owner of this ability can delete an object
+    struct DeleteAbility has copy, drop, store {
+        self: address,
+    }
+
+    /// The owner of this ability can generate the object's signer
+    struct SignerAbility has copy, drop, store {
+        self: address,
+    }
+
+    /// This implies that this individual owns this object
+    struct OwnerAbility has drop, store {
+        self: address,
+    }
+
+    /// This implies that the individual owns this object and it contains a specific type / resource
+    struct TypedOwnerAbility<phantom T: key> has drop, store {
+        self: address,
+    }
+
+    // This is heterogeneous store of owned objects
+    struct ObjectStore has key {
+        inner: table::Table<address, OwnerAbility>,
+        deposits: event::EventHandle<DepositEvent>,
+        withdraws: event::EventHandle<WithdrawEvent>,
+    }
+
+    struct DepositEvent has drop, store {
+        object_id: address,
+    }
+
+    struct WithdrawEvent has drop, store {
+        object_id: address,
+    }
+
+    public entry fun init_store(account: &signer) {
+        move_to(
+            account,
+            ObjectStore {
+                inner: table::new(),
+                deposits: account::new_event_handle(account),
+                withdraws: account::new_event_handle(account),
+            },
+        )
+    }
+
+    public fun withdraw(account: &signer, addr: address): OwnerAbility acquires ObjectStore {
+        let object_store = borrow_global_mut<ObjectStore>(signer::address_of(account));
+        event::emit_event(&mut object_store.withdraws, WithdrawEvent { object_id: addr });
+        table::remove(&mut object_store.inner, addr)
+    }
+
+    public fun deposit(account: &signer, object: OwnerAbility) acquires ObjectStore {
+        let object_store = borrow_global_mut<ObjectStore>(signer::address_of(account));
+        let object_addr = owner_ability_address(&object);
+        event::emit_event(&mut object_store.deposits, DepositEvent { object_id: object_addr });
+        table::add(&mut object_store.inner, object_addr, object);
+    }
+
+    public fun deposit_typed<T: key>(
+        account: &signer,
+        object: TypedOwnerAbility<T>,
+    ) acquires ObjectStore {
+        deposit(account, to_owner_ability(object))
+    }
+
+    public fun contains(account: &signer, addr: address): bool acquires ObjectStore {
+        let object_store = borrow_global<ObjectStore>(signer::address_of(account));
+        table::contains(&object_store.inner, addr)
+    }
+
+    /// Object addresses are equal to the sha3_256([creator addres | seed | 0xFE]).
+    public fun create_object_id(source: &address, seed: vector<u8>): address {
```

Comment:
Can we keep the term (object id vs object address) consistent across the entire module?